baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/Slf4jLevelCheck.java

```diff
@@ -118,6 +122,23 @@ private ConditionVisitor() {
         public Optional<MethodInvocationTree> visitParenthesized(ParenthesizedTree node, VisitorState state) {
             return node.getExpression().accept(this, state);
         }
+
+        // It's relatively common to do a quick check (often equality or null check) along with a level check.
+        // We only support expressions with a single level check matched with '&&' to keep things simple, as this
+        // is the most common case.
+        @Override
+        public Optional<MethodInvocationTree> visitBinary(BinaryTree node, VisitorState state) {
+            if (node.getKind() != Tree.Kind.CONDITIONAL_AND) {
+                return Optional.empty();
+            }
+            Optional<MethodInvocationTree> lhs = node.getLeftOperand().accept(this, state);
+            Optional<MethodInvocationTree> rhs = node.getRightOperand().accept(this, state);
+            // If there are level checks on both sides, bail
+            if (lhs.isPresent() && rhs.isPresent()) {
```

Comment:
Are these `.isPresent()` calls definitely safe even when the visitCatch on 160 returns `null`? Does it always make a reduce call under the hood??

Reply:
160 is a method on the MostSevereLogStatementScanner TreeScanner, this is the ConditionVisitor (SimpleVisitor).

I wanted to use Optional.empty in the treescanner, but it's a bit more difficult because it doesn't allow a configurable default value.